### PR TITLE
 Remove old support libraries

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

 Remove old support libraries

## Related PRs or issues

[AB#85280](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/85280)

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
     private static final String[] LOCATIONS = new String[]{"androidx.test.platform.app.InstrumentationRegistry",
             "androidx.test.InstrumentationRegistry",
-            "android.support.test.InstrumentationRegistry"};
+            "androidx.test.platform.app.InstrumentationRegistry"};
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java:19]: Two identical strings "androidx.test.platform.app.InstrumentationRegistry" here.

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java [hunk 2]
```diff
@@ -10,10 +10,11 @@
 import android.content.Context;
 import android.database.Cursor;
 import android.database.sqlite.SQLiteQueryBuilder;
-import android.support.annotation.NonNull;
-import android.support.test.InstrumentationRegistry;
-import android.support.test.filters.MediumTest;
-import android.support.test.runner.AndroidJUnit4;
+import androidx.annotation.NonNull;
+
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java:14]: empty line

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/PersistenceAndroidTest.java [hunk 3]
```diff
@@ -7,9 +7,10 @@
 
 import android.annotation.SuppressLint;
 import android.content.Context;
-import android.support.test.InstrumentationRegistry;
-import android.support.test.filters.SmallTest;
-import android.support.test.runner.AndroidJUnit4;
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
+
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/PersistenceAndroidTest.java:11]: empty line

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java [hunk 4]
```diff
@@ -11,9 +11,10 @@
 import android.database.Cursor;
 import android.database.sqlite.SQLiteDatabase;
 import android.database.sqlite.SQLiteQueryBuilder;
-import android.support.test.InstrumentationRegistry;
-import android.support.test.filters.SmallTest;
-import android.support.test.runner.AndroidJUnit4;
+
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java:14]: empty line

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 4.1.2 (Under active development)
 
+### App Center
+
+* **[Fix]** Remove old support libraries.
```
Comment (MatkovIvan) [inline CHANGELOG.md:7]: The wording is not clear at all. What the impact for customers?

File: apps/sasquatch/build.gradle [hunk 6]
```diff
@@ -81,10 +81,9 @@ repositories {
 }
 
 dependencies {
-    implementation "com.android.support:support-v4:${rootProject.ext.supportLibVersion}"
-    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
-    implementation "com.android.support:customtabs:${rootProject.ext.supportLibVersion}"
-    implementation "com.android.support:recyclerview-v7:${rootProject.ext.supportLibVersion}"
+    implementation "androidx.appcompat:appcompat:1.0.2"
+    implementation "androidx.browser:browser:1.0.0"
+    implementation "androidx.recyclerview:recyclerview:1.1.0"
```
Comment (MatkovIvan) [inline apps/sasquatch/build.gradle:86]: Does it mean that all applications that use SDK have to change it? Is it breaking change?

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: apps/sasquatch/build.gradle (hunk 6 above)
Comment (AnastasiaKubova) [reply apps/sasquatch/build.gradle:86]: Nope, these changes need only for the correct passing of our CI steps

**Review by russelarms — DISMISSED** [review]

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java [hunk 7]
```diff
@@ -15,8 +15,7 @@
 public class InstrumentationRegistryHelper {
 
     private static final String[] LOCATIONS = new String[]{"androidx.test.platform.app.InstrumentationRegistry",
-            "androidx.test.InstrumentationRegistry",
-            "android.support.test.InstrumentationRegistry"};
+            "androidx.test.InstrumentationRegistry"};
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java:18]: It's a detection code, it should not be removed

File: sdk/build.gradle [hunk 8]
```diff
@@ -78,25 +78,27 @@ subprojects {
     }
 
     dependencies {
-        compileOnly "com.android.support:support-annotations:${ext.supportLibVersion}"
-        javadocDeps "com.android.support:support-annotations:${ext.supportLibVersion}"
+        def annotationVersion = '1.2.0'
```
Comment (MatkovIvan) [inline sdk/build.gradle:81]: We have separated file for version constants

File: sdk/build.gradle [hunk 9]
```diff
@@ -78,25 +78,27 @@ subprojects {
     }
 
     dependencies {
-        compileOnly "com.android.support:support-annotations:${ext.supportLibVersion}"
-        javadocDeps "com.android.support:support-annotations:${ext.supportLibVersion}"
```
Comment (MatkovIvan) [inline sdk/build.gradle:82]: `ext.supportLibVersion` is unused now

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**